astropy/visualization/interval.py

```diff
@@ -222,6 +220,8 @@ class ZScaleInterval(BaseInterval):
         .. versionchanged:: 5.2
             ``n_samples`` replaces the deprecated ``nsamples`` argument,
             which will be removed in the future.
```

Comment:
Should the stuff about 5.2 be removed?

Reply:
I would lean towards "no", this is still valuable information for anyone consulting our docs wondering when these changes happened.
If we really don't want to leave it around, I think we shouldn't add a new note at all, but I have a preference for keeping both.

Reply:
Can we combine them?  E.g., keep the 7.0 note, but have something like:
"`n_samples` replaced `nsamples` (deprecated in 5.2)"

Reply:
Apparently Python completely replaces the notices about deprecations at the end of the deprecation period and the final notes do not mention when the deprecation period started. See for example `asyncio.gather()` in [3.9](https://docs.python.org/3.9/library/asyncio-task.html#asyncio.gather) and in [3.10](https://docs.python.org/3.10/library/asyncio-task.html#asyncio.gather) or `ftplib.FTP_TLS` in [3.11](https://docs.python.org/3.11/library/ftplib.html#ftp-tls-objects) and [3.12](https://docs.python.org/3.12/library/ftplib.html#ftp-tls-objects).

Reply:
In that case my preference now goes to dropping those notes too. We can afford it since our documentation is versioned anyway, so they won't be *completely* gone.